src/test/java/hudson/util/VersionNumberTest.java

```diff
@@ -71,4 +71,20 @@ public void testTimestamps() {
         assertFalse(new VersionNumber("2.0.3-20170207.105042-1").isNewerThan(new VersionNumber("2.0.3-SNAPSHOT")));
         assertFalse(new VersionNumber("2.0.3-20170207.105042-1").isOlderThan(new VersionNumber("2.0.3-SNAPSHOT")));
     }
+
+    public void testDigit() {
+        assertEquals(32, new VersionNumber("2.32.3.1-SNAPSHOT").getDigitAt(1));
+        assertEquals(3, new VersionNumber("2.32.3.1-SNAPSHOT").getDigitAt(2));
+        assertEquals(1, new VersionNumber("2.32.3.1-SNAPSHOT").getDigitAt(3));
+        assertEquals(-1, new VersionNumber("2.32.3.1-SNAPSHOT").getDigitAt(4));
+        assertEquals(2, new VersionNumber("2.7.22.0.2").getDigitAt(4));
+        assertEquals(3, new VersionNumber("2.7.22.0.3-SNAPSHOT").getDigitAt(4));
+        assertEquals(-1, new VersionNumber("2.0.3-20170207.105042-1").getDigitAt(4));
```

Comment:
Is this correct really?
Depending on how it's parsed I could argue that all three of `20170207.105042`, `20170207` and `105042` are numbers/digits. Or what defines a "numeric component"?